src/pages/Main/components/OptionInput.jsx

```diff
@@ -0,0 +1,54 @@
+import React from 'react';
+import styled from 'styled-components';
+import variables from '../../../styles/variables';
+import { BsCheckCircleFill, BsCheckCircle } from 'react-icons/bs';
+import { useSearchParams } from 'react-router-dom';
+
+const OptionInput = ({
+  item: { id, category, content },
+  setCheckedElement,
+  checkedElement,
+}) => {
+  const handleRadioButton = e => {
+    setCheckedElement(Number(e.target.value));
+    searchParams.set(category, e.target.value);
+    setSearchParams(searchParams);
+  };
+
+  const [searchParams, setSearchParams] = useSearchParams();
+
+  return (
+    <OptionWrapper>
+      <input
+        type="radio"
+        value={id}
+        checked={checkedElement === id}
+        onChange={handleRadioButton}
+      />
+      {checkedElement === id ? (
+        <BsCheckCircleFill className="radioIcon" />
+      ) : (
+        <BsCheckCircle />
+      )}
+      <OptionText>{content}</OptionText>
```

Comment:
따로 스타일링을 해줄 필요가 없다면
```suggestion
      {content}
```
로만 작성가능하겠습니다!